Include/cpython/pyatomic_gcc.h

```diff
@@ -489,12 +489,12 @@ _Py_atomic_store_ullong_relaxed(unsigned long long *obj,
 // --- _Py_atomic_load_ptr_acquire / _Py_atomic_store_ptr_release ------------
 
 static inline void *
-_Py_atomic_load_ptr_acquire(const void *obj)
+_Py_atomic_load_ptr_acquire(void *obj)
 { return (void *)__atomic_load_n((void **)obj, __ATOMIC_ACQUIRE); }
 
 static inline uintptr_t
 _Py_atomic_load_uintptr_acquire(const uintptr_t *obj)
-{ return (uintptr_t)__atomic_load_n((uintptr_t *)obj, __ATOMIC_ACQUIRE); }
+{ return (uintptr_t)__atomic_load_n(obj, __ATOMIC_ACQUIRE); }
```

Comment:
I don't understand this change. Shouldn't you rather remove the `const` from the signature?

Reply:
Without this change, the compiler emits a warning.

Reply:
Yes, but isn't it because the object being passed to `_Py_atomic_load_uintptr_acquire` is a `const uintptr_t *`? 

Now that I look a bit around, all the signatures take a `const T *` but I'm not sure if it's better to remove the inner cast or to fix the cast itself. 

For instance, 

```C
static inline void *
_Py_atomic_load_ptr_acquire(const void *obj)
{ return (void *)__atomic_load_n((void **)obj, __ATOMIC_ACQUIRE); }
```

could be fixed as 

```C
static inline void *
_Py_atomic_load_ptr_acquire(const void *obj)
{ return (void *)__atomic_load_n((void * const*)obj, __ATOMIC_ACQUIRE); }
```

instead of changing the signature from `const void *` to `void *` (also, the cast on the return type should not needed I think). What do you think of that? at least, we would keep the same logic for signatures.

Reply:
> Yes, but isn't it because the object being passed to _Py_atomic_load_uintptr_acquire is a const uintptr_t *?

The compiler warns if you cast from "const TYPE" to "TYPE". My change removes the cast. It just works.

> instead of changing the signature from const void * to void *

`void*` is special. I gave up trying to fix the warning, so I changed the parameters type of the 3 "pointer" functions instead.

Did you try your proposed fix? When I tried, it didn't work for me.

Reply:
Well yes, it worked for me actually.. that's why I suggested it. I changed this:

```C
// L491 in pyatomic_gcc.h
static inline void *
_Py_atomic_load_ptr_acquire(const void *obj)
{ return (void *)__atomic_load_n((void **)obj, __ATOMIC_ACQUIRE); }
``` 

into

```C
static inline void *
_Py_atomic_load_ptr_acquire(const void *obj)
{ return __atomic_load_n((void * const*)obj, __ATOMIC_ACQUIRE); }
```

and I did not have any warning for this specific function (there were more but not for this one).

Reply:
Maybe I tried `const void * const *`, I don't recall.

Anyway, I wrote a new PR instead: PR gh-121055.

Reply:
Thank you for your nice advice, it worked well at the end ;-)